### ISSUE TITLE
Remove `uses-permission` from AndroidManifest.xml

### ISSUE THREAD

**Issue by sulaysumaria** [briankabiro/react-native-get-sms-android#34]
#### Problem Statement:
I am using `react-native-get-sms-android` for reading sms only. Still my app asks permission for `WRITE_SMS` and `SEND_SMS`, even though I have not asked for that permission.

#### Cause:
The permission specified in the `AndroidManifest.xml` of this plugin.

#### Solution:
Remove the permissions from this plugin's `AndroidManifest.xml` and let developers add it to their app's Manifest so that there are no permissions required that the developer is not aware of.

#### Workaround:
Add this to your app's `android/app/src/main/AndroidManifest.xml` and it will remove unnecessary permission.
```xml
<uses-permission android:name="android.permission.WRITE_SMS" tools:node="remove" />
<uses-permission android:name="android.permission.SEND_SMS" tools:node="remove" />
```

Also add `xmlns:tools` to your `manifest` tag to work:
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="com.example">
```

**Comment by sulaysumaria** [comment]
Also change the README accordingly. As it guides to add permissions manually to AndroidManifest.xml, which is not actually required. It should be changed to add the specific permission that a specific function requires.

**Comment by mikehardy** [comment]
Seems like a reasonable idea. Why not propose a PR to match?

**Comment by sulaysumaria** [comment]
Sure thing. I am new to react-native, so was just going through on some blog posts on how to develop plugins for react-native so that I don't accidentally break any other things. I will create a PR as soon as I am ready with the change. Maybe within 24 hours.

**Comment by briankabiro** [comment]
Thanks @sulaysumaria. We'd both be happy to offer advice on this.

**Comment by sulaysumaria** [comment]
I realized what I did here. 😆 .

I looked at `index.js` and there are just two lines!! I will leat every reader to view the file by himself.

Basically what this package does is exports the `react-native`'s `NativeModules.Sms` and adds permissions for `READ_SMS`, `WRITE_SMS` and `SEND_SMS` to `AndroidManifest.xml`, so that developer don't need to add it to app's manifest.

And what I suggested was exactly opposite of that. If developers add the permissions to the app's manifest, there is no use of this package anymore!!

@briankabiro , @mikehardy thanks for maintaining this package. There are two things to learn hereL

1. Don't blindly include any package into your project's dependencies, there might not be any code at all!!!
2. Take a look at what the code does, maybe we can do that in our code without adding the package (like importing NativeModules from react-native)

Also I am removing this package from my app's dependencies and replacing it with `react-native`'s `NativeModules.Sms`.

I am not creating the PR for this change as it will break the whole point of maintaining this in the first place. Again thanks for this package. 👍
